### PR TITLE
orphan uri clean up

### PR DESCRIPTION
**Pull request by ymatsuda**
@eishay @raymondng @andrewconner 

I didn't know that OrphanCleaner had been disabled for a long time. It seems we accumulated many orphans in NormalizedURI table.

This change replaces old orphan clean up logic. The new logic is driven from Bookmark table using its sequence number. The idea is we only check URIs in changed bookmarks. URLRepo is no longer used.

Also, ScrapeInfo clean up is moved to ScrapeSchedulerPluginImpl. We no longer have a nightly clean up task. Instead, it is done on-the-fly as we schedule scraping tasks.

Cron job is not set up yet.

The high watermark (highest processed sequence number) is in memory, so we forget the progress and have to start from scratch after server restart. It is better to persist it in DB or ZK.

### CODE REVIEW

**Comment by andrewconner** [comment]
Great. Didn't know it had been disabled, good catch.

**Comment by raymondng** [comment]
Nice. Thanks!

File: kifi-backend/modules/shoebox/app/com/keepit/integrity/DataIntegrity.scala [hunk 1]
```diff
@@ -61,49 +48,68 @@ private[integrity] class DataIntegrityActor @Inject() (
...
-          oldScrapeInfos = oldScrapeInfos.+:(si)
+    var seq = bookmarkSequenceNumber
+    var done = false
+    while (!done) {
+      db.readWrite{ implicit s =>
+        val bookmarks = bookmarkRepo.getBookmarksChanged(seq, 100)
+        done = bookmarks.isEmpty
+        bookmarks.foreach{ bookmark =>
+          val uri = try {
+            Some(nuriRepo.get(bookmark.uriId))
```
Comment (eishay) [inline kifi-backend/modules/shoebox/app/com/keepit/integrity/DataIntegrity.scala:76]: can use `nuriRepo.getOpt` ?

File: kifi-backend/modules/shoebox/app/com/keepit/integrity/DataIntegrity.scala [hunk 2]
```diff
@@ -61,49 +48,68 @@ private[integrity] class DataIntegrityActor @Inject() (
...
+          } catch { case e: Throwable =>
+            airbrake.notify("error getting NormalizedURI in OrphanCleaner", e)
+            None
+          }
+          uri.foreach{ u =>
```
Comment (eishay) [inline kifi-backend/modules/shoebox/app/com/keepit/integrity/DataIntegrity.scala:81]: we should have a test for that, its not trivial

File: kifi-backend/modules/shoebox/app/com/keepit/integrity/DataIntegrity.scala [hunk 3]
```diff
@@ -43,17 +34,13 @@ private[integrity] case object Cron
...
     case CleanOrphans =>
-      db.readWrite { implicit session =>
-        // This cleans up cases when we have a normalizedUri, but no Url. This *only* happens when we renormalize, so does not need to happen every night.
-        //orphanCleaner.cleanNormalizedURIs(false)
-        orphanCleaner.cleanScrapeInfo(false)
-      }
+      // This cleans up cases when we have a normalizedUri, but no Url. This *only* happens when we renormalize, so does not need to happen every night.
+      //orphanCleaner.cleanNormalizedURIs(false)
     case Cron =>
       if (currentDateTime.hourOfDay().get() == 21) // 9pm PST
```
Comment (eishay) [inline kifi-backend/modules/shoebox/app/com/keepit/integrity/DataIntegrity.scala:45]: do we still want to do it once a day at something like `currentDateTime.hourOfDay().get() == 21` ?
if so i'll update the code to use proper cron syntax

File: kifi-backend/modules/shoebox/app/com/keepit/integrity/DataIntegrity.scala (hunk 1 above)
Comment (ymatsuda) [inline kifi-backend/modules/shoebox/app/com/keepit/integrity/DataIntegrity.scala:76]: getOpt takes an external id.

**Pull request merged**